### PR TITLE
Stopped increasing the cache size to accomodate large images.

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

Reverted e4e8ab4 and replaced it with logic that just doesn't cache images that don't fit in the cache.  The issue that predicated increasing the cache size dynamically in the first place doesn't manifest anymore.  There is more work to be done to make the cache even better but this gets rid of this erroneous logic.

## Related Issues

https://github.com/flutter/flutter/issues/45643

## Tests

I added the following tests:

image_cache_test :: 'Do not cache large images'

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [x] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [x] I wrote a migration guide: https://github.com/flutter/website/pull/3453

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by dnfield — APPROVED** [review]
LGTM, but should wait for migration guide to land.

**Comment by goderbauer** [comment]
What's the status of this PR? Is it ready to land?

**Comment by gaaclarke** [comment]
It's ready to land.  I put a pin into it so we didn't land it while people
were OOO.  Yesterday I sent the breaking change announcement over email.
The breaking change announcement on the website already landed.  I'm just
waiting for the light to turn green before I merge.
a

**Pull request merged**
